tests/test_interpolation.py

```diff
@@ -709,3 +710,31 @@ def test_empty_stack() -> None:
     """
     with pytest.raises(GrammarParseError):
         grammar_parser.parse("ab}", lexer_mode="VALUE_MODE")
+
+
+@pytest.mark.parametrize("ref", ["missing", "invalid"])
+def test_invalid_intermediate_result_when_not_throwing(
+    ref: str, restore_resolvers: Any
+) -> None:
+    """
+    Test the handling of missing / resolution failures in nested interpolations.
+
+    The main goal of this test is to make sure that the resolution of an interpolation
+    is stopped immediately when a missing / resolution failure occurs, even if
+    `throw_on_missing` and `throw_on_resolution_failure` are set to False.
+    When this happens while dereferencing a node, the result should be `None`.
+    """
+
+    def fail_if_called(x: Any) -> None:
+        assert False
+
+    OmegaConf.register_new_resolver("fail_if_called", fail_if_called)
+    cfg = OmegaConf.create({"x": f"${{fail_if_called:${{{ref}}}}}", "missing": "???"})
```

Comment:
woa.
maybe don't use f strings here.
I have no idea which { belongs to the interpolation and which one to the fstring.

Reply:
Hehe indeed, it looked ok in VSCode because it adds colors, but definitely confusing on github... % to the rescue in fa39026